### PR TITLE
fix(snuba): Snuba timeseries serializer was adding an extra entry to the payload response

### PR DESCRIPTION
**Pull request by dashed**
An extra time series entry was being added to the Snuba time series payload response. This is making graphs look wonky:

Screen Shot 2020-03-03 at 7 13 47 PM
Screen Shot 2020-03-03 at 7 13 08 PM

## TODO

- [x] fix tests

### CODE REVIEW

File: tests/sentry/api/serializers/test_incident.py [hunk 1]
```diff
@@ -35,8 +35,8 @@ def test_simple(self):
         assert result["dateDetected"] == incident.date_detected
         assert result["dateCreated"] == incident.date_added
         assert result["dateClosed"] == incident.date_closed
-        assert len(result["eventStats"]["data"]) == 62
-        assert [data[1] for data in result["eventStats"]["data"]] == [[]] * 62
+        assert len(result["eventStats"]["data"]) == 61
```
Comment (dashed) [inline tests/sentry/api/serializers/test_incident.py]: @iProgramStuff @wedamija @billyvg  Can you folks verify if this is right? I'm not familiar with the internals of incidents.

Comment (dashed) [reply tests/sentry/api/serializers/test_incident.py]: Looks like this is right.

File: src/sentry/api/serializers/snuba.py [hunk 2]
```diff
@@ -126,7 +126,7 @@ def value_from_row(row, tagkey):
 
 def zerofill(data, start, end, rollup):
     rv = []
-    start = ((int(to_timestamp(start)) // rollup) * rollup) - rollup
+    start = (int(to_timestamp(start)) // rollup) * rollup
```
Comment (markstory) [inline src/sentry/api/serializers/snuba.py]: This will also change the bar graphs shown on issue list and issue details.

Comment (dashed) [reply src/sentry/api/serializers/snuba.py]: @markstory I think this change is correct. 

I didn't see any Percy regression for the issue list and issue details page. But there were some snapshot changes for the Incidents pages.

I'm assuming any frontend components will need to be updated to compensate this change.

File: tests/snuba/api/endpoints/test_organization_events_stats.py [hunk 3]
```diff
@@ -64,17 +64,17 @@ def test_simple(self):
...
             },
             format="json",
         )
 
         assert response.status_code == 200, response.content
         assert [attrs for time, attrs in response.data["data"]] == [
-            [],
             [{"count": 1}],
             [{"count": 2}],
+            [{"count": 0}],
```
Comment (billyvg) [inline tests/snuba/api/endpoints/test_organization_events_stats.py]: Why are there 3 entries here?

File: tests/sentry/api/serializers/test_incident.py (hunk 1 above)
Comment (billyvg) [reply tests/sentry/api/serializers/test_incident.py]: Yeah seems consistent with the other changes.

File: tests/snuba/api/endpoints/test_organization_events_stats.py (hunk 3 above)
Comment (dashed) [reply tests/snuba/api/endpoints/test_organization_events_stats.py]: @billyvg Good question.

I updated this test to ensure returned data points are only within the requested time range. For example:

If start= 12 PM, end = 2 PM, and interval = 1 hour, then we should expect 3 data points. These points refer to 12 PM, 1 PM, and 2 PM buckets.

**Review by billyvg — APPROVED** [review]

**Pull request merged**